lib/oidc/renewTokensWithRefresh.ts

```diff
@@ -11,41 +11,32 @@
  *
  */
 import { AuthSdkError } from '../errors';
-import http from '../http';
 import { getOAuthUrls } from './util/oauth';
 import { OktaAuth, TokenParams, RefreshToken, Tokens } from '../types';
 import { handleOAuthResponse } from './handleOAuthResponse';
+import { postRefreshToken } from './endpoints/token';
+import { getDefaultTokenParams } from './util';
 
 export async function renewTokensWithRefresh(
   sdk: OktaAuth,
   tokenParams: TokenParams,
   refreshTokenObject: RefreshToken
 ): Promise<Tokens> {
-  var clientId = sdk.options.clientId;
+  const { clientId } = sdk.options;
   if (!clientId) {
-    throw new AuthSdkError('A clientId must be specified in the OktaAuth constructor to revoke a token');
+    throw new AuthSdkError('A clientId must be specified in the OktaAuth constructor to renew tokens');
   }
 
-  var urls = getOAuthUrls(sdk, tokenParams);
+  const renewTokenParams = {
+    clientId,
+  };
+  const tokenResponse = await postRefreshToken(sdk, renewTokenParams, refreshTokenObject);
 
-
-  const response = await http.httpRequest(sdk, {
-    url: refreshTokenObject.tokenUrl,
-    method: 'POST',
-    withCredentials: false,
-    headers: {
-      'Content-Type': 'application/x-www-form-urlencoded',
-    },
-
-    args: Object.entries({
-      client_id: clientId, // eslint-disable-line camelcase
-      grant_type: 'refresh_token', // eslint-disable-line camelcase
-      scope: refreshTokenObject.scopes.join(' '),
-      refresh_token: refreshTokenObject.refreshToken, // eslint-disable-line camelcase
-    }).map(function ([name, value]) {
-      return name + '=' + encodeURIComponent(value);
-    }).join('&'),
-  });
-  return handleOAuthResponse(sdk, tokenParams, response, urls).then(res => res.tokens);
-
-}
\ No newline at end of file
+  const { scopes } = getDefaultTokenParams(sdk);
+  const handleResponseParams = {
+    scopes
```

Comment:
SDK's `scopes` configuration is [used](https://github.com/okta/okta-auth-js/blob/master/lib/oidc/handleOAuthResponse.ts#L65) for building parsed token objects, however `scopes` is also available in `/token` response - would it make sense to use it instead?

Reply:
It does probably make sense to use the value from the response rather than config. `scopes` can be overridden on the call used to obtain the tokens, so it could differ from the configured value.

Reply:
Done